runtime/src/serde_snapshot/types.rs

```diff
@@ -0,0 +1,48 @@
+use {
+    serde::{de, Deserialize, Deserializer},
+    std::{fmt, marker::PhantomData},
+};
+
+/// Snapshot serde-safe AccountsLtHash
+#[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
+#[derive(Debug)]
+pub struct SerdeAccountsLtHash(
+    // serde only has array support up to 32 elements; anything larger needs to be handled manually
+    // see https://github.com/serde-rs/serde/issues/1937 for more information
+    pub [u16; 1024],
+);
+
+impl<'de> Deserialize<'de> for SerdeAccountsLtHash {
```

Comment:
yuck. I do not like to review this. @t-nelson thoughts?

Reply:
if we aren't willing to review logic, we have no business using a dependency that to hide it from us. especially the case for a backport

Reply:
in this case the dependency was already used by runtime, just not by accounts-db, as I understand it. So, yes, we can review it, but it sure feels nicer reviewing the simple thing than this. But, you certainly have a point. The fact we're adding a dependency feels more like a code modularization byproduct than it does a new risk.

Reply:
From what I review, "serde_as" procedure macro is already being used for packet class, which is u8[1232]. And here we are just using it again in a similar way for u16[1024]. Therefore, I think it should be fine.

https://github.com/anza-xyz/agave/blob/489f483e1d7b30ef114e0123994818b2accfa389/sdk/src/packet.rs#L93

Also, looking at the source repo for "serde_as", there is already sufficient tests coverage of "serde_as" for the fixed size array case that we used here. 
https://github.com/jonasbb/serde_with/blob/f1b79f27cac7545513c4f14f7c295db2b843cc4f/serde_with/tests/serde_as/lib.rs#L321

Here, instead of using "serde_as", we basically implement just deserialize specially for the u16[1024] by hand, which I believe is correct. And I trust that books has already tested it manually. But as far as the test for this manually written the code, the coverage may not be as good as those tests already in the "serde_as" crate. 

Therefore, IMHO, I slightly prefer use "serde_as". But I am fine with the hand implementation too.